src/routes/safe/container/Hooks/useLoadStore.jsx

```diff
@@ -0,0 +1,31 @@
+// @flow
+import { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import loadAddressBookFromStorage from '~/logic/addressBook/store/actions/loadAddressBookFromStorage'
+import addViewedSafe from '~/logic/currentSession/store/actions/addViewedSafe'
+import fetchLatestMasterContractVersion from '~/routes/safe/store/actions/fetchLatestMasterContractVersion'
+import fetchSafe from '~/routes/safe/store/actions/fetchSafe'
+import fetchTransactions from '~/routes/safe/store/actions/fetchTransactions'
+import { safeParamAddressFromStateSelector } from '~/routes/safe/store/selectors'
+
+export const useLoadStore = (setSafeLoaded: Function) => {
+  const dispatch = useDispatch()
+  const safeAddress = useSelector(safeParamAddressFromStateSelector)
+
+  useEffect(() => {
+    const fetchData = () => {
+      if (safeAddress) {
+        dispatch(fetchLatestMasterContractVersion())
```

Comment:
why do we need to fetch latest master contract version on every safe change? this seems like something that can be placed in `src/index.js` to be fetched once on app startup

Reply:
We need this because `fetchLatestMasterContractVersion` should be called before calling `fetchSafe` of course this should be done just one time but we can't avoid a race condition if we split the calls

Reply:
you can check if this has been fetched before fetching the safe